listeners/test-client.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var Mocha = require('mocha');
+var fs = require('fs');
+var path = require('path');
+var Q = require('q');
+
+class TestClient {
+
+    constructor () {
+      var self = this;
+      self._testResult = Q.defer();
+    }
+
+    _getTestResult(){
```

Comment:
Since this is called from outside the class (main.bt) it's public and should be named getTestResult, without the underscore